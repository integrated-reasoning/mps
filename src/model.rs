use crate::types::{Parser, RowType, Rows};
use color_eyre::{eyre::eyre, Result};
use hashbrown::HashMap;

#[derive(Debug, Default, Clone, PartialEq, Eq)]
pub struct Model {
  pub name: String,
  pub row_types: RowTypeMap,
}

#[derive(Debug, Default, Clone, PartialEq, Eq)]
pub struct RowTypeMap(HashMap<String, RowType>);

impl TryFrom<Parser<'_, f32>> for Model {
  type Error = color_eyre::Report;

  fn try_from(parsed: Parser<f32>) -> Result<Self> {
    let row_types = RowTypeMap::try_from(parsed.rows)?;
    Ok(Model {
      name: parsed.name.to_string(),
      row_types,
    })
  }
}

impl TryFrom<Rows<'_>> for RowTypeMap {
  type Error = color_eyre::Report;

  fn try_from(rows: Rows<'_>) -> Result<Self> {
    let mut row_types = HashMap::new();
    for r in rows {
      match row_types.insert(r.row_name.to_string(), r.row_type.clone()) {
        Some(row_type) => Err(eyre!(format!(
          "conflicting row type information for {}: found {:?} and {:?}",
          r.row_name,
          r.row_type.clone(),
          row_type
        ))),
        None => Ok(()),
      }?;
    }
    Ok(RowTypeMap(row_types))
  }
}

mod tests {
  use super::*;

  #[test]
  fn test_try_from_afiro() -> Result<()> {
    let (_, parsed) =
      Parser::<f32>::parse(include_str!("../tests/data/netlib/afiro"))?;
    Model::try_from(parsed)?;
    Ok(())
  }

  #[test]
  fn test_model_conflicting_rows_line() -> Result<()> {
    let (_, parsed) = Parser::<f32>::parse(include_str!(
      "../tests/data/should_fail/conflicting_rows_line"
    ))?;
    let error =
      eyre!("conflicting row type information for R09: found Leq and Eq");
    match Model::try_from(parsed) {
      Ok(_) => panic!(),
      Err(e) => assert_eq!(e.to_string(), error.to_string()),
    };
    Ok(())
  }

  #[test]
  fn test_rowtypemap_conflicting_rows_line() -> Result<()> {
    let (_, parsed) = Parser::<f32>::parse(include_str!(
      "../tests/data/should_fail/conflicting_rows_line"
    ))?;
    let error =
      eyre!("conflicting row type information for R09: found Leq and Eq");
    match RowTypeMap::try_from(parsed.rows) {
      Ok(_) => panic!(),
      Err(e) => assert_eq!(e.to_string(), error.to_string()),
    };
    Ok(())
  }
}
