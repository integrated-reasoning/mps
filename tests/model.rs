mod tests {
  use color_eyre::Result;
  use mps::model::Model;
  use mps::Parser;

  #[test]
  fn test_model_from_agg() -> Result<()> {
    let parsed =
      Parser::<f32>::parse(include_str!("../tests/data/netlib/agg"))?;
    let model = Model::try_from(parsed)?;
    insta::assert_yaml_snapshot!(model);
    Ok(())
  }

  #[test]
  fn test_model_from_ship04l() -> Result<()> {
    let parsed =
      Parser::<f32>::parse(include_str!("../tests/data/netlib/ship04l"))?;
    let model = Model::try_from(parsed)?;
    insta::assert_yaml_snapshot!(model);
    Ok(())
  }

  #[test]
  fn test_model_from_d2q06c() -> Result<()> {
    let parsed =
      Parser::<f32>::parse(include_str!("../tests/data/netlib/d2q06c"))?;
    let model = Model::try_from(parsed)?;
    insta::assert_yaml_snapshot!(model);
    Ok(())
  }

  #[test]
  fn test_model_from_e226() -> Result<()> {
    let parsed =
      Parser::<f32>::parse(include_str!("../tests/data/netlib/e226"))?;
    let model = Model::try_from(parsed)?;
    insta::assert_yaml_snapshot!(model);
    Ok(())
  }

  #[test]
  fn test_model_from_nl25fv47() -> Result<()> {
    let parsed =
      Parser::<f32>::parse(include_str!("../tests/data/netlib/25fv47"))?;
    let model = Model::try_from(parsed)?;
    insta::assert_yaml_snapshot!(model);
    Ok(())
  }

  #[test]
  fn test_model_from_bore3d() -> Result<()> {
    let parsed =
      Parser::<f32>::parse(include_str!("../tests/data/netlib/bore3d"))?;
    let model = Model::try_from(parsed)?;
    insta::assert_yaml_snapshot!(model);
    Ok(())
  }

  #[test]
  fn test_model_from_ganges() -> Result<()> {
    let parsed =
      Parser::<f32>::parse(include_str!("../tests/data/netlib/ganges"))?;
    let model = Model::try_from(parsed)?;
    insta::assert_yaml_snapshot!(model);
    Ok(())
  }

  #[test]
  fn test_model_from_adlittle() -> Result<()> {
    let parsed =
      Parser::<f32>::parse(include_str!("../tests/data/netlib/adlittle"))?;
    let model = Model::try_from(parsed)?;
    insta::assert_yaml_snapshot!(model);
    Ok(())
  }

  #[test]
  fn test_model_from_forplan() -> Result<()> {
    let parsed =
      Parser::<f32>::parse(include_str!("../tests/data/netlib/forplan"))?;
    let model = Model::try_from(parsed)?;
    insta::assert_yaml_snapshot!(model);
    Ok(())
  }

  #[test]
  fn test_model_from_sc205() -> Result<()> {
    let parsed =
      Parser::<f32>::parse(include_str!("../tests/data/netlib/sc205"))?;
    let model = Model::try_from(parsed)?;
    insta::assert_yaml_snapshot!(model);
    Ok(())
  }

  #[test]
  fn test_model_from_nl80bau3b() -> Result<()> {
    let parsed =
      Parser::<f32>::parse(include_str!("../tests/data/netlib/80bau3b"))?;
    let model = Model::try_from(parsed)?;
    insta::assert_yaml_snapshot!(model);
    Ok(())
  }

  #[test]
  fn test_model_from_scrs8() -> Result<()> {
    let parsed =
      Parser::<f32>::parse(include_str!("../tests/data/netlib/scrs8"))?;
    let model = Model::try_from(parsed)?;
    insta::assert_yaml_snapshot!(model);
    Ok(())
  }

  #[test]
  fn test_model_from_wood1p() -> Result<()> {
    let parsed =
      Parser::<f32>::parse(include_str!("../tests/data/netlib/wood1p"))?;
    let model = Model::try_from(parsed)?;
    insta::assert_yaml_snapshot!(model);
    Ok(())
  }

  #[test]
  fn test_model_from_boeing1() -> Result<()> {
    let parsed =
      Parser::<f32>::parse(include_str!("../tests/data/netlib/boeing1"))?;
    let model = Model::try_from(parsed)?;
    insta::assert_yaml_snapshot!(model);
    Ok(())
  }

  #[test]
  fn test_model_from_kb2() -> Result<()> {
    let parsed =
      Parser::<f32>::parse(include_str!("../tests/data/netlib/kb2"))?;
    let model = Model::try_from(parsed)?;
    insta::assert_yaml_snapshot!(model);
    Ok(())
  }

  #[test]
  fn test_model_from_ship08s() -> Result<()> {
    let parsed =
      Parser::<f32>::parse(include_str!("../tests/data/netlib/ship08s"))?;
    let model = Model::try_from(parsed)?;
    insta::assert_yaml_snapshot!(model);
    Ok(())
  }

  #[test]
  fn test_model_from_scfxm1() -> Result<()> {
    let parsed =
      Parser::<f32>::parse(include_str!("../tests/data/netlib/scfxm1"))?;
    let model = Model::try_from(parsed)?;
    insta::assert_yaml_snapshot!(model);
    Ok(())
  }

  #[test]
  fn test_model_from_agg2() -> Result<()> {
    let parsed =
      Parser::<f32>::parse(include_str!("../tests/data/netlib/agg2"))?;
    let model = Model::try_from(parsed)?;
    insta::assert_yaml_snapshot!(model);
    Ok(())
  }

  #[test]
  fn test_model_from_finnis() -> Result<()> {
    let parsed =
      Parser::<f32>::parse(include_str!("../tests/data/netlib/finnis"))?;
    let model = Model::try_from(parsed)?;
    insta::assert_yaml_snapshot!(model);
    Ok(())
  }

  #[test]
  fn test_model_from_dfl001() -> Result<()> {
    let parsed =
      Parser::<f32>::parse(include_str!("../tests/data/netlib/dfl001"))?;
    let model = Model::try_from(parsed)?;
    insta::assert_yaml_snapshot!(model);
    Ok(())
  }

  #[test]
  fn test_model_from_pilot87() -> Result<()> {
    let parsed =
      Parser::<f32>::parse(include_str!("../tests/data/netlib/pilot87"))?;
    let model = Model::try_from(parsed)?;
    insta::assert_yaml_snapshot!(model);
    Ok(())
  }

  #[test]
  fn test_model_from_sctap1() -> Result<()> {
    let parsed =
      Parser::<f32>::parse(include_str!("../tests/data/netlib/sctap1"))?;
    let model = Model::try_from(parsed)?;
    insta::assert_yaml_snapshot!(model);
    Ok(())
  }

  #[test]
  fn test_model_from_agg3() -> Result<()> {
    let parsed =
      Parser::<f32>::parse(include_str!("../tests/data/netlib/agg3"))?;
    let model = Model::try_from(parsed)?;
    insta::assert_yaml_snapshot!(model);
    Ok(())
  }

  #[test]
  fn test_model_from_grow7() -> Result<()> {
    let parsed =
      Parser::<f32>::parse(include_str!("../tests/data/netlib/grow7"))?;
    let model = Model::try_from(parsed)?;
    insta::assert_yaml_snapshot!(model);
    Ok(())
  }

  #[test]
  fn test_model_from_scorpion() -> Result<()> {
    let parsed =
      Parser::<f32>::parse(include_str!("../tests/data/netlib/scorpion"))?;
    let model = Model::try_from(parsed)?;
    insta::assert_yaml_snapshot!(model);
    Ok(())
  }

  #[test]
  fn test_model_from_maros() -> Result<()> {
    let parsed =
      Parser::<f32>::parse(include_str!("../tests/data/netlib/maros"))?;
    let model = Model::try_from(parsed)?;
    insta::assert_yaml_snapshot!(model);
    Ok(())
  }

  #[test]
  fn test_model_from_shell() -> Result<()> {
    let parsed =
      Parser::<f32>::parse(include_str!("../tests/data/netlib/shell"))?;
    let model = Model::try_from(parsed)?;
    insta::assert_yaml_snapshot!(model);
    Ok(())
  }

  #[test]
  fn test_model_from_greenbeb() -> Result<()> {
    let parsed =
      Parser::<f32>::parse(include_str!("../tests/data/netlib/greenbeb"))?;
    let model = Model::try_from(parsed)?;
    insta::assert_yaml_snapshot!(model);
    Ok(())
  }

  #[test]
  fn test_model_from_sc50b() -> Result<()> {
    let parsed =
      Parser::<f32>::parse(include_str!("../tests/data/netlib/sc50b"))?;
    let model = Model::try_from(parsed)?;
    insta::assert_yaml_snapshot!(model);
    Ok(())
  }

  #[test]
  fn test_model_from_recipe() -> Result<()> {
    let parsed =
      Parser::<f32>::parse(include_str!("../tests/data/netlib/recipe"))?;
    let model = Model::try_from(parsed)?;
    insta::assert_yaml_snapshot!(model);
    Ok(())
  }

  #[test]
  fn test_model_from_sierra() -> Result<()> {
    let parsed =
      Parser::<f32>::parse(include_str!("../tests/data/netlib/sierra"))?;
    let model = Model::try_from(parsed)?;
    insta::assert_yaml_snapshot!(model);
    Ok(())
  }

  #[test]
  fn test_model_from_scagr25() -> Result<()> {
    let parsed =
      Parser::<f32>::parse(include_str!("../tests/data/netlib/scagr25"))?;
    let model = Model::try_from(parsed)?;
    insta::assert_yaml_snapshot!(model);
    Ok(())
  }

  #[test]
  fn test_model_from_modszk1() -> Result<()> {
    let parsed =
      Parser::<f32>::parse(include_str!("../tests/data/netlib/modszk1"))?;
    let model = Model::try_from(parsed)?;
    insta::assert_yaml_snapshot!(model);
    Ok(())
  }

  #[test]
  fn test_model_from_ship12l() -> Result<()> {
    let parsed =
      Parser::<f32>::parse(include_str!("../tests/data/netlib/ship12l"))?;
    let model = Model::try_from(parsed)?;
    insta::assert_yaml_snapshot!(model);
    Ok(())
  }

  #[test]
  fn test_model_from_stair() -> Result<()> {
    let parsed =
      Parser::<f32>::parse(include_str!("../tests/data/netlib/stair"))?;
    let model = Model::try_from(parsed)?;
    insta::assert_yaml_snapshot!(model);
    Ok(())
  }

  #[test]
  fn test_model_from_cycle() -> Result<()> {
    let parsed =
      Parser::<f32>::parse(include_str!("../tests/data/netlib/cycle"))?;
    let model = Model::try_from(parsed)?;
    insta::assert_yaml_snapshot!(model);
    Ok(())
  }

  #[test]
  fn test_model_from_sc105() -> Result<()> {
    let parsed =
      Parser::<f32>::parse(include_str!("../tests/data/netlib/sc105"))?;
    let model = Model::try_from(parsed)?;
    insta::assert_yaml_snapshot!(model);
    Ok(())
  }

  #[test]
  fn test_model_from_pilot_ja() -> Result<()> {
    let parsed =
      Parser::<f32>::parse(include_str!("../tests/data/netlib/pilot.ja"))?;
    let model = Model::try_from(parsed)?;
    insta::assert_yaml_snapshot!(model);
    Ok(())
  }

  #[test]
  fn test_model_from_beaconfd() -> Result<()> {
    let parsed =
      Parser::<f32>::parse(include_str!("../tests/data/netlib/beaconfd"))?;
    let model = Model::try_from(parsed)?;
    insta::assert_yaml_snapshot!(model);
    Ok(())
  }

  #[test]
  fn test_model_from_czprob() -> Result<()> {
    let parsed =
      Parser::<f32>::parse(include_str!("../tests/data/netlib/czprob"))?;
    let model = Model::try_from(parsed)?;
    insta::assert_yaml_snapshot!(model);
    Ok(())
  }

  #[test]
  fn test_model_from_pilot_we() -> Result<()> {
    let parsed =
      Parser::<f32>::parse(include_str!("../tests/data/netlib/pilot.we"))?;
    let model = Model::try_from(parsed)?;
    insta::assert_yaml_snapshot!(model);
    Ok(())
  }

  #[test]
  fn test_model_from_standgub() -> Result<()> {
    let parsed =
      Parser::<f32>::parse(include_str!("../tests/data/netlib/standgub"))?;
    let model = Model::try_from(parsed)?;
    insta::assert_yaml_snapshot!(model);
    Ok(())
  }

  #[test]
  fn test_model_from_standmps() -> Result<()> {
    let parsed =
      Parser::<f32>::parse(include_str!("../tests/data/netlib/standmps"))?;
    let model = Model::try_from(parsed)?;
    insta::assert_yaml_snapshot!(model);
    Ok(())
  }

  #[test]
  fn test_model_from_scsd8() -> Result<()> {
    let parsed =
      Parser::<f32>::parse(include_str!("../tests/data/netlib/scsd8"))?;
    let model = Model::try_from(parsed)?;
    insta::assert_yaml_snapshot!(model);
    Ok(())
  }

  #[test]
  fn test_model_from_woodw() -> Result<()> {
    let parsed =
      Parser::<f32>::parse(include_str!("../tests/data/netlib/woodw"))?;
    let model = Model::try_from(parsed)?;
    insta::assert_yaml_snapshot!(model);
    Ok(())
  }

  #[test]
  fn test_model_from_scsd6() -> Result<()> {
    let parsed =
      Parser::<f32>::parse(include_str!("../tests/data/netlib/scsd6"))?;
    let model = Model::try_from(parsed)?;
    insta::assert_yaml_snapshot!(model);
    Ok(())
  }

  #[test]
  fn test_model_from_scsd1() -> Result<()> {
    let parsed =
      Parser::<f32>::parse(include_str!("../tests/data/netlib/scsd1"))?;
    let model = Model::try_from(parsed)?;
    insta::assert_yaml_snapshot!(model);
    Ok(())
  }

  #[test]
  fn test_model_from_share2b() -> Result<()> {
    let parsed =
      Parser::<f32>::parse(include_str!("../tests/data/netlib/share2b"))?;
    let model = Model::try_from(parsed)?;
    insta::assert_yaml_snapshot!(model);
    Ok(())
  }

  #[test]
  fn test_model_from_gfrd_pnc() -> Result<()> {
    let parsed =
      Parser::<f32>::parse(include_str!("../tests/data/netlib/gfrd-pnc"))?;
    let model = Model::try_from(parsed)?;
    insta::assert_yaml_snapshot!(model);
    Ok(())
  }

  #[test]
  fn test_model_from_bnl2() -> Result<()> {
    let parsed =
      Parser::<f32>::parse(include_str!("../tests/data/netlib/bnl2"))?;
    let model = Model::try_from(parsed)?;
    insta::assert_yaml_snapshot!(model);
    Ok(())
  }

  #[test]
  fn test_model_from_stocfor2() -> Result<()> {
    let parsed =
      Parser::<f32>::parse(include_str!("../tests/data/netlib/stocfor2"))?;
    let model = Model::try_from(parsed)?;
    insta::assert_yaml_snapshot!(model);
    Ok(())
  }

  #[test]
  fn test_model_from_nesm() -> Result<()> {
    let parsed =
      Parser::<f32>::parse(include_str!("../tests/data/netlib/nesm"))?;
    let model = Model::try_from(parsed)?;
    insta::assert_yaml_snapshot!(model);
    Ok(())
  }

  #[test]
  fn test_model_from_share1b() -> Result<()> {
    let parsed =
      Parser::<f32>::parse(include_str!("../tests/data/netlib/share1b"))?;
    let model = Model::try_from(parsed)?;
    insta::assert_yaml_snapshot!(model);
    Ok(())
  }

  #[test]
  fn test_model_from_ship04s() -> Result<()> {
    let parsed =
      Parser::<f32>::parse(include_str!("../tests/data/netlib/ship04s"))?;
    let model = Model::try_from(parsed)?;
    insta::assert_yaml_snapshot!(model);
    Ok(())
  }

  #[test]
  fn test_model_from_grow15() -> Result<()> {
    let parsed =
      Parser::<f32>::parse(include_str!("../tests/data/netlib/grow15"))?;
    let model = Model::try_from(parsed)?;
    insta::assert_yaml_snapshot!(model);
    Ok(())
  }

  #[test]
  fn test_model_from_maros_r7() -> Result<()> {
    let parsed =
      Parser::<f32>::parse(include_str!("../tests/data/netlib/maros-r7"))?;
    let model = Model::try_from(parsed)?;
    insta::assert_yaml_snapshot!(model);
    Ok(())
  }

  #[test]
  fn test_model_from_blend() -> Result<()> {
    let parsed =
      Parser::<f32>::parse(include_str!("../tests/data/netlib/blend"))?;
    let model = Model::try_from(parsed)?;
    insta::assert_yaml_snapshot!(model);
    Ok(())
  }

  #[test]
  fn test_model_from_lotfi() -> Result<()> {
    let parsed =
      Parser::<f32>::parse(include_str!("../tests/data/netlib/lotfi"))?;
    let model = Model::try_from(parsed)?;
    insta::assert_yaml_snapshot!(model);
    Ok(())
  }

  #[test]
  fn test_model_from_standata() -> Result<()> {
    let parsed =
      Parser::<f32>::parse(include_str!("../tests/data/netlib/standata"))?;
    let model = Model::try_from(parsed)?;
    insta::assert_yaml_snapshot!(model);
    Ok(())
  }

  #[test]
  fn test_model_from_d6cube() -> Result<()> {
    let parsed =
      Parser::<f32>::parse(include_str!("../tests/data/netlib/d6cube"))?;
    let model = Model::try_from(parsed)?;
    insta::assert_yaml_snapshot!(model);
    Ok(())
  }

  #[test]
  fn test_model_from_degen3() -> Result<()> {
    let parsed =
      Parser::<f32>::parse(include_str!("../tests/data/netlib/degen3"))?;
    let model = Model::try_from(parsed)?;
    insta::assert_yaml_snapshot!(model);
    Ok(())
  }

  #[test]
  fn test_model_from_capri() -> Result<()> {
    let parsed =
      Parser::<f32>::parse(include_str!("../tests/data/netlib/capri"))?;
    let model = Model::try_from(parsed)?;
    insta::assert_yaml_snapshot!(model);
    Ok(())
  }

  #[test]
  fn test_model_from_grow22() -> Result<()> {
    let parsed =
      Parser::<f32>::parse(include_str!("../tests/data/netlib/grow22"))?;
    let model = Model::try_from(parsed)?;
    insta::assert_yaml_snapshot!(model);
    Ok(())
  }

  #[test]
  fn test_model_from_etamacro() -> Result<()> {
    let parsed =
      Parser::<f32>::parse(include_str!("../tests/data/netlib/etamacro"))?;
    let model = Model::try_from(parsed)?;
    insta::assert_yaml_snapshot!(model);
    Ok(())
  }

  #[test]
  fn test_model_from_ship08l() -> Result<()> {
    let parsed =
      Parser::<f32>::parse(include_str!("../tests/data/netlib/ship08l"))?;
    let model = Model::try_from(parsed)?;
    insta::assert_yaml_snapshot!(model);
    Ok(())
  }

  #[test]
  fn test_model_from_afiro() -> Result<()> {
    let parsed =
      Parser::<f32>::parse(include_str!("../tests/data/netlib/afiro"))?;
    let model = Model::try_from(parsed)?;
    insta::assert_yaml_snapshot!(model);
    Ok(())
  }

  #[test]
  fn test_model_from_degen2() -> Result<()> {
    let parsed =
      Parser::<f32>::parse(include_str!("../tests/data/netlib/degen2"))?;
    let model = Model::try_from(parsed)?;
    insta::assert_yaml_snapshot!(model);
    Ok(())
  }

  #[test]
  fn test_model_from_boeing2() -> Result<()> {
    let parsed =
      Parser::<f32>::parse(include_str!("../tests/data/netlib/boeing2"))?;
    let model = Model::try_from(parsed)?;
    insta::assert_yaml_snapshot!(model);
    Ok(())
  }

  #[test]
  fn test_model_from_fit1d() -> Result<()> {
    let parsed =
      Parser::<f32>::parse(include_str!("../tests/data/netlib/fit1d"))?;
    let model = Model::try_from(parsed)?;
    insta::assert_yaml_snapshot!(model);
    Ok(())
  }

  #[test]
  fn test_model_from_scfxm2() -> Result<()> {
    let parsed =
      Parser::<f32>::parse(include_str!("../tests/data/netlib/scfxm2"))?;
    let model = Model::try_from(parsed)?;
    insta::assert_yaml_snapshot!(model);
    Ok(())
  }

  #[test]
  fn test_model_from_sctap3() -> Result<()> {
    let parsed =
      Parser::<f32>::parse(include_str!("../tests/data/netlib/sctap3"))?;
    let model = Model::try_from(parsed)?;
    insta::assert_yaml_snapshot!(model);
    Ok(())
  }

  #[test]
  fn test_model_from_fit1p() -> Result<()> {
    let parsed =
      Parser::<f32>::parse(include_str!("../tests/data/netlib/fit1p"))?;
    let model = Model::try_from(parsed)?;
    insta::assert_yaml_snapshot!(model);
    Ok(())
  }

  #[test]
  fn test_model_from_pilot() -> Result<()> {
    let parsed =
      Parser::<f32>::parse(include_str!("../tests/data/netlib/pilot"))?;
    let model = Model::try_from(parsed)?;
    insta::assert_yaml_snapshot!(model);
    Ok(())
  }

  #[test]
  fn test_model_from_fit2d() -> Result<()> {
    let parsed =
      Parser::<f32>::parse(include_str!("../tests/data/netlib/fit2d"))?;
    let model = Model::try_from(parsed)?;
    insta::assert_yaml_snapshot!(model);
    Ok(())
  }

  #[test]
  fn test_model_from_sctap2() -> Result<()> {
    let parsed =
      Parser::<f32>::parse(include_str!("../tests/data/netlib/sctap2"))?;
    let model = Model::try_from(parsed)?;
    insta::assert_yaml_snapshot!(model);
    Ok(())
  }

  #[test]
  fn test_model_from_scfxm3() -> Result<()> {
    let parsed =
      Parser::<f32>::parse(include_str!("../tests/data/netlib/scfxm3"))?;
    let model = Model::try_from(parsed)?;
    insta::assert_yaml_snapshot!(model);
    Ok(())
  }

  #[test]
  fn test_model_from_brandy() -> Result<()> {
    let parsed =
      Parser::<f32>::parse(include_str!("../tests/data/netlib/brandy"))?;
    let model = Model::try_from(parsed)?;
    insta::assert_yaml_snapshot!(model);
    Ok(())
  }

  #[test]
  fn test_model_from_greenbea() -> Result<()> {
    let parsed =
      Parser::<f32>::parse(include_str!("../tests/data/netlib/greenbea"))?;
    let model = Model::try_from(parsed)?;
    insta::assert_yaml_snapshot!(model);
    Ok(())
  }

  #[test]
  fn test_model_from_tuff() -> Result<()> {
    let parsed =
      Parser::<f32>::parse(include_str!("../tests/data/netlib/tuff"))?;
    let model = Model::try_from(parsed)?;
    insta::assert_yaml_snapshot!(model);
    Ok(())
  }

  #[test]
  fn test_model_from_sc50a() -> Result<()> {
    let parsed =
      Parser::<f32>::parse(include_str!("../tests/data/netlib/sc50a"))?;
    let model = Model::try_from(parsed)?;
    insta::assert_yaml_snapshot!(model);
    Ok(())
  }

  #[test]
  fn test_model_from_vtp_base() -> Result<()> {
    let parsed =
      Parser::<f32>::parse(include_str!("../tests/data/netlib/vtp.base"))?;
    let model = Model::try_from(parsed)?;
    insta::assert_yaml_snapshot!(model);
    Ok(())
  }

  #[test]
  fn test_model_from_pilotnov() -> Result<()> {
    let parsed =
      Parser::<f32>::parse(include_str!("../tests/data/netlib/pilotnov"))?;
    let model = Model::try_from(parsed)?;
    insta::assert_yaml_snapshot!(model);
    Ok(())
  }

  #[test]
  fn test_model_from_ship12s() -> Result<()> {
    let parsed =
      Parser::<f32>::parse(include_str!("../tests/data/netlib/ship12s"))?;
    let model = Model::try_from(parsed)?;
    insta::assert_yaml_snapshot!(model);
    Ok(())
  }

  #[test]
  fn test_model_from_seba() -> Result<()> {
    let parsed =
      Parser::<f32>::parse(include_str!("../tests/data/netlib/seba"))?;
    let model = Model::try_from(parsed)?;
    insta::assert_yaml_snapshot!(model);
    Ok(())
  }

  #[test]
  fn test_model_from_fffff800() -> Result<()> {
    let parsed =
      Parser::<f32>::parse(include_str!("../tests/data/netlib/fffff800"))?;
    let model = Model::try_from(parsed)?;
    insta::assert_yaml_snapshot!(model);
    Ok(())
  }

  #[test]
  fn test_model_from_israel() -> Result<()> {
    let parsed =
      Parser::<f32>::parse(include_str!("../tests/data/netlib/israel"))?;
    let model = Model::try_from(parsed)?;
    insta::assert_yaml_snapshot!(model);
    Ok(())
  }

  #[test]
  fn test_model_from_perold() -> Result<()> {
    let parsed =
      Parser::<f32>::parse(include_str!("../tests/data/netlib/perold"))?;
    let model = Model::try_from(parsed)?;
    insta::assert_yaml_snapshot!(model);
    Ok(())
  }

  #[test]
  fn test_model_from_pilot4() -> Result<()> {
    let parsed =
      Parser::<f32>::parse(include_str!("../tests/data/netlib/pilot4"))?;
    let model = Model::try_from(parsed)?;
    insta::assert_yaml_snapshot!(model);
    Ok(())
  }

  #[test]
  fn test_parse_scagr7() -> Result<()> {
    insta::assert_yaml_snapshot!(Parser::<f32>::parse(include_str!(
      "../tests/data/netlib/scagr7"
    ))?);
    Ok(())
  }

  #[test]
  fn test_parse_bandm() -> Result<()> {
    insta::assert_yaml_snapshot!(Parser::<f32>::parse(include_str!(
      "../tests/data/netlib/bandm"
    ))?);
    Ok(())
  }

  #[test]
  fn test_parse_bnl1() -> Result<()> {
    insta::assert_yaml_snapshot!(Parser::<f32>::parse(include_str!(
      "../tests/data/netlib/bnl1"
    ))?);
    Ok(())
  }

  #[test]
  fn test_parse_stocfor1() -> Result<()> {
    insta::assert_yaml_snapshot!(Parser::<f32>::parse(include_str!(
      "../tests/data/netlib/stocfor1"
    ))?);
    Ok(())
  }
}
