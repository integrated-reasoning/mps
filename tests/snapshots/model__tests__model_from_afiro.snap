---
source: tests/model.rs
expression: model
---
name: AFIRO
row_types:
  X47: Leq
  X17: Leq
  X48: Leq
  X49: Leq
  X51: Leq
  R12: Eq
  X19: Leq
  R23: Eq
  R09: Eq
  X20: Leq
  COST: Nr
  X27: Leq
  R22: Eq
  X44: Leq
  X05: Leq
  X18: Leq
  R13: Eq
  X50: Leq
  R19: Eq
  X21: Leq
  R20: Eq
  X46: Leq
  R10: Eq
  X40: Leq
  X41: Leq
  X42: Leq
  X43: Leq
  X45: Leq
values:
  ? - X45
    - X35
  : 2.279
  ? - X45
    - X13
  : 2.429
  ? - R12
    - X09
  : -1
  ? - R10
    - X04
  : 1
  ? - R12
    - X15
  : 1
  ? - X51
    - X16
  : 1
  ? - X48
    - X24
  : -1
  ? - R13
    - X06
  : -1.06
  ? - X47
    - X15
  : -1
  ? - R19
    - X25
  : 1
  ? - X18
    - X07
  : 1
  ? - X19
    - X12
  : -1
  ? - R13
    - X08
  : -0.96
  ? - X20
    - X13
  : -1
  ? - X45
    - X25
  : -1
  ? - X43
    - X31
  : 1
  ? - X40
    - X32
  : -1
  ? - R23
    - X37
  : 1
  ? - X45
    - X11
  : 2.386
  ? - X40
    - X28
  : 1
  ? - R10
    - X01
  : -1.06
  ? - X45
    - X33
  : 2.219
  ? - X50
    - X26
  : 1
  ? - X46
    - X03
  : -1
  ? - X20
    - X09
  : 1
  ? - X49
    - X06
  : 0.301
  ? - R22
    - X38
  : 1
  ? - X18
    - X11
  : -1
  ? - X49
    - X07
  : 0.313
  ? - R22
    - X28
  : -0.43
  ? - X45
    - X34
  : 2.249
  ? - X47
    - X31
  : 0.107
  ? - X27
    - X22
  : 1
  ? - R22
    - X30
  : -0.39
  ? - R12
    - X14
  : 1
  ? - X21
    - X02
  : -1
  ? - X19
    - X08
  : 1
  ? - R09
    - X01
  : -1
  ? - R22
    - X31
  : -0.37
  ? - R23
    - X31
  : 1
  ? - X45
    - X10
  : 2.364
  ? - X48
    - X01
  : 0.301
  ? - X47
    - X28
  : 0.109
  ? - X45
    - X32
  : 2.191
  ? - X49
    - X08
  : 0.313
  ? - R23
    - X28
  : 1
  ? - R23
    - X30
  : 1
  ? - R13
    - X07
  : -1.06
  ? - R23
    - X39
  : 1
  ? - R12
    - X08
  : -1
  ? - X45
    - X12
  : 2.408
  ? - X46
    - X22
  : 0.109
  ? - X17
    - X10
  : -1
  ? - R20
    - X26
  : 1
  ? - R20
    - X22
  : -0.43
  ? - X51
    - X38
  : 1
  ? - R13
    - X16
  : 1
  ? - X42
    - X34
  : -1
  ? - COST
    - X02
  : -0.4
  ? - X49
    - X09
  : 0.326
  ? - R19
    - X24
  : 1
  ? - X17
    - X06
  : 1
  ? - COST
    - X14
  : -0.32
  ? - R19
    - X23
  : 1
  ? - X05
    - X01
  : 1
  ? - R22
    - X29
  : -0.43
  ? - R09
    - X03
  : 1
  ? - R23
    - X29
  : 1
  ? - R12
    - X07
  : -1
  ? - X41
    - X33
  : -1
  ? - R09
    - X02
  : 1
  ? - X49
    - X37
  : -1
  ? - X41
    - X29
  : 1
  ? - R23
    - X36
  : -1
  ? - X44
    - X36
  : 1.4
  ? - X50
    - X04
  : 1
  ? - X42
    - X30
  : 1
  ? - COST
    - X36
  : -0.48
  ? - X21
    - X14
  : 1.4
  ? - R13
    - X09
  : -0.86
  ? - X44
    - X23
  : -1
  ? - COST
    - X39
  : 10
  ? - X47
    - X30
  : 0.108
  ? - COST
    - X23
  : -0.6
  ? - R19
    - X22
  : -1
  ? - R12
    - X06
  : -1
  ? - X47
    - X29
  : 0.108
  ? - X43
    - X35
  : -1
rhs:
  B:
    X51: 300
    X05: 80
    X17: 80
    X27: 500
    X40: 500
    X50: 310
    R23: 44
bounds: {}
ranges: {}

